### ISSUE TITLE
Consider adding a mode where the generated code is dependency-free

### ISSUE THREAD

**Issue by eliben** [deepmap/oapi-codegen#303]
Currently the generated code has a dependency on `pkg/runtime` in this module, mostly to support `Bind*` functions.

While generated methods of the wrapper know exactly which types are needed in Go for the params, they call fairly generic `Bind*` functions that take these params via `interface{}`. This seems a bit backward - if we already have a concrete type, why bind it through a generic function that uses runtime dispatch?

It should be possible for the generated code to directly bind parameters - given a known concrete type it should be only a few lines of code per parameter. This would be faster, but more importantly **dependency-free** - the generated code would only be dependent on Echo (or other framework used).

**Comment by deepmap-marcinr** [comment]
Yes, I've been meaning to do that, but it's _really_ tricky. I've got a branch where it kinda works, but I have to break out all the special cases we have into their own functions and insert those into the spec. We'd also need to inline a whole lot of string manipulation functions. One day :)

**Comment by mromaszewicz** [comment]
The feature request for dependency-free generated code is explicitly addressed in the V3 experimental codegen (oapi-codegen-exp). The V3 README states: 'V2 generated code relied on github.com/oapi-codegen/runtime for parameter binding and styling. This was a complaint from lots of people due to various audit requirements. V3 embeds all necessary helper functions and helper types into the spec.' V2 stable still requires the runtime import unconditionally (imports.tmpl:29), but this is the intended architecture for V2. The enhancement has landed as a design principle of V3.

Closing as fixed. If you can still reproduce on a recent release, please reopen with the exact spec and version used.
